src/NServiceBus.RabbitMQ/Sending/BasicPropertiesExtensions.cs

```diff
@@ -32,12 +32,15 @@ public static void Fill(this IBasicProperties properties, OutgoingMessage messag
 
             properties.Headers = message.Headers.ToDictionary(p => p.Key, p => (object)p.Value);
 
-            if (message.Headers.ContainsKey(NServiceBus.Headers.EnclosedMessageTypes))
+            string messageTypesHeader;
+            if (message.Headers.TryGetValue(NServiceBus.Headers.EnclosedMessageTypes, out messageTypesHeader))
             {
-                properties.Type = message.Headers[NServiceBus.Headers.EnclosedMessageTypes].Split(new[]
+                var index = messageTypesHeader.IndexOf(',');
+
+                if (index > 0)
                 {
-                    ','
-                }, StringSplitOptions.RemoveEmptyEntries).FirstOrDefault();
+                    properties.Type = messageTypesHeader.Substring(0, index);
+                }
```

Comment:
`BasicPropertiesExtensions.Fill` gets called when NSB is sending a message, so native integration isn't relevant here. For this to not have a comma, core would have to fundamentally change the value being stored in this header.